azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java

```diff
@@ -28,7 +40,34 @@
      * @throws AzureLoginFailureException when there are some errors during login.
      */
     public static AzureCredential oAuthLogin(AzureEnvironment env) throws AzureLoginFailureException {
-        throw new UnsupportedOperationException("Not implemented");
+        try {
+            final WebAppServer webAppServer = new WebAppServer();
+            final String redirectUrl = webAppServer.getUrl();
+            final URI redirectUri = new URI(redirectUrl);
+            try {
+                final String authorizationUrl = authorizationUrl(env, redirectUrl);
+                try {
+                    Desktop.getDesktop().browse(new URL(authorizationUrl).toURI());
+                    System.out.println(AUTH_WITH_OAUTH);
+                } catch (Exception ex) {
```

Comment:
We need to record the error and send telemetry to understand what users are experiencing.

Reply:
The telemetry will not be in this PR, it is considered to be done later(after BI spec review is done)